src/libstd/fs.rs

```diff
@@ -844,6 +844,9 @@ pub fn symlink_metadata<P: AsRef<Path>>(path: P) -> io::Result<Metadata> {
 
 /// Rename a file or directory to a new name.
 ///
+/// Moving a file or directory is also a rename, in a sense, and so this
+/// function is not limited a single directory.
```

Comment:
No, talking in IRC, the example I used was the behavior they weren't familiar with.